backend/learnings/views.py

```diff
@@ -247,13 +247,20 @@ class GetUserActivityLog(APIView):
     authentication_classes = [TokenAuthentication]
 
     def get(self, request, page, username):
-        user = EUser.objects.get(username=username)
         if page == "dashboard":
-            activities = UserActivityLog.objects.filter(user_id=user).order_by("-created_at")
-        else:
             activities = UserActivityLog.objects.filter(
-                activity_description="quiz", user_id=user
+                user_id__username=username, activity_description="follow"
             ).order_by("-created_at")
+            for other_user in activities:
+                activity = UserActivityLog.objects.filter(
+                    user_id__username=other_user.follow_id.username
+                ).order_by("-created_at")
+                if activity.exists():
+                    activities = activities | activity
```

Comment:
try this po, since theres a lot of queries and this looks cleaner if this will work :bow: 
```
users = Follow.objects.filter(follower_id__username=username).values_list('following_id__username', flat=True)
users = list(users) // not sure if this is needed
users = users.append(username)

activities = UserActivityLog.objects.filter(user_id__username__in=users).order_by("-created_at")
```

if the above did not work try this as well
```
users = Follow.objects.filter(follower_id__username=username)
users = map(lambda user: user.get('username'), users)
users = users.append(username)

activities = UserActivityLog.objects.filter(user_id__username__in=users).order_by("-created_at")
```